strange issue related to nested `all`

**V version:** 
latest

**OS:** 
win10 

**What did you do?**
Thanks to @yuyi98 's work on nested `all`, I'm trying to replace a for loop with `all`, forming a nested `all` case there.
Position: scanner.v, from around line 930.
(Lines to be replaced are shown commented here.)
```V
						if s.last_lt >= 0 && s.pos - s.last_lt < 100 {
							typs := s.text[s.last_lt + 1..s.pos].split(',').map(it.trim_space().trim_right('>').after(']'))
							//if any typ is neither Type nor builtin, then the case is non-generic

                                                        // this is the added new line
							is_generic = typs.all(it.len > 0 && ((it[0].is_capital() && it[1..].bytes().all(it.is_alnum() || it == `_`)) || it in ast.builtin_type_names))

							//for typ in typs {
							//	if typ.len == 0
							//		|| (!(typ[0].is_capital() && typ[1..].bytes().all(it.is_alnum()
							//		|| it == `_`)) && typ !in ast.builtin_type_names) {
							//		is_generic = false
							//		break
							//	}
							//}
						} else {
							is_generic = false
						}
```

**What did you expect to see?**
It's an equivalent replacement so running `make -tcc` should still work.
 
**What did you see instead?**
Bootstrap failed
```
Building V...
 > Attempting to build v_win.c with TCC
 > Compiling with .\v.exe self
failed    cmd: C:\Programming\vlang\v.exe -cc C:\PROGRA~4\vlang\THIRDP~1\tcc\tcc.exe -o v2 cmd/v
failed   code: 1
==================
C:/Users/Roy/AppData/Local/Temp/v/v2.15590162803657400086.tmp.c:29068: warning: assignment from incompatible pointer type
C:/Users/Roy/AppData/Local/Temp/v/v2.15590162803657400086.tmp.c:59654: error: 'it' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

**Analysis**
Try compile the new v_win.c generated I get
```
vc/v_win.c:59654: error: 'it' undeclared
``` 
So indeed there's an issue. The code stuck nearby is 
```C
						for (int _t51 = 0; _t51 < _t50_len; ++_t51) {
							string it = ((string*) _t50_orig.data)[_t51];
							string ti = string_after(string_trim_right(string_trim_space(it), _SLIT(">")), _SLIT("]"));
							array_push((array*)&_t50, &ti);
						}
						
						Array_string typs = _t50;
						Array_byte _t54_orig = string_bytes(string_substr(it, 1, it.len));  //Issue's here!
						int _t54_len = _t54_orig.len;
						bool _t54 = true;
						for (int _t55 = 0; _t55 < _t54_len; ++_t55) {
							byte it = ((byte*) _t54_orig.data)[_t55];
							if (!(byte_is_alnum(it) || it == '_')) {
								_t54 = false;
			break;
		}
						}
```

--- a/vlib/v/fmt/fmt.v
+++ b/vlib/v/fmt/fmt.v
@@ -1724,7 +1724,7 @@ pub fn (mut f Fmt) ident(node ast.Ident) {
 		}
 	}
 	f.write_language_prefix(node.language)
-	if node.name == 'it' && f.it_name != '' && f.inside_lambda == 0 { // allow `it` in lambdas
+	if node.name == 'it' && f.it_name != '' && f.in_lambda_depth == 0 { // allow `it` in lambdas
 		f.write(f.it_name)
 	} else if node.kind == .blank_ident {
 		f.write('_')

--- a/vlib/v/fmt/fmt.v
+++ b/vlib/v/fmt/fmt.v
@@ -43,7 +43,7 @@ pub mut:
 	use_short_fn_args  bool
 	single_line_fields bool   // should struct fields be on a single line
 	it_name            string // the name to replace `it` with
-	inside_lambda      bool
+	in_lambda_depth    int
 	inside_const       bool
 	is_mbranch_expr    bool // match a { x...y { } }
 	fn_scope           &ast.Scope = voidptr(0)
@@ -1539,9 +1539,9 @@ pub fn (mut f Fmt) call_expr(node ast.CallExpr) {
 	}
 	if node.is_method {
 		if node.name in ['map', 'filter', 'all', 'any'] {
-			f.inside_lambda = true
+			f.in_lambda_depth++
 			defer {
-				f.inside_lambda = false
+				f.in_lambda_depth--
 			}
 		}
 		if node.left is ast.Ident {
@@ -1724,7 +1724,7 @@ pub fn (mut f Fmt) ident(node ast.Ident) {
 		}
 	}
 	f.write_language_prefix(node.language)
-	if node.name == 'it' && f.it_name != '' && !f.inside_lambda { // allow `it` in lambdas
+	if node.name == 'it' && f.it_name != '' && f.inside_lambda == 0 { // allow `it` in lambdas
 		f.write(f.it_name)
 	} else if node.kind == .blank_ident {
 		f.write('_')

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -196,7 +196,7 @@ fn (mut g Gen) gen_array_map(node ast.CallExpr) {
 			}
 		}
 		ast.CallExpr {
-			if expr.name in ['map', 'filter'] {
+			if expr.name in ['map', 'filter', 'all', 'any'] {
 				is_embed_map_filter = true
 				g.stmt_path_pos << g.out.len
 			}
@@ -387,7 +387,7 @@ fn (mut g Gen) gen_array_filter(node ast.CallExpr) {
 			}
 		}
 		ast.CallExpr {
-			if expr.name in ['map', 'filter'] {
+			if expr.name in ['map', 'filter', 'all', 'any'] {
 				is_embed_map_filter = true
 				g.stmt_path_pos << g.out.len
 			}
@@ -678,7 +678,7 @@ fn (mut g Gen) gen_array_any(node ast.CallExpr) {
 			}
 		}
 		ast.CallExpr {
-			if expr.name in ['map', 'filter'] {
+			if expr.name in ['map', 'filter', 'all', 'any'] {
 				is_embed_map_filter = true
 				g.stmt_path_pos << g.out.len
 			}
@@ -761,7 +761,7 @@ fn (mut g Gen) gen_array_all(node ast.CallExpr) {
 			}
 		}
 		ast.CallExpr {
-			if expr.name in ['map', 'filter'] {
+			if expr.name in ['map', 'filter', 'all', 'any'] {
 				is_embed_map_filter = true
 				g.stmt_path_pos << g.out.len
 			}

--- a/vlib/v/scanner/scanner.v
+++ b/vlib/v/scanner/scanner.v
@@ -926,20 +926,14 @@ fn (mut s Scanner) text_scan() token.Token {
 						// here we set the limit 100 which should be nice for real cases
 						// e.g. ...Bar<int, []Foo<int>, Baz_, [20]f64, map[string][]bool>> =>
 						// <int, Baz_, [20]f64, map[string][]bool => int, Baz_, f64, bool
-						mut is_generic := true
-						if s.last_lt >= 0 && s.pos - s.last_lt < 100 {
+						is_generic := if s.last_lt >= 0 && s.pos - s.last_lt < 100 {
 							typs := s.text[s.last_lt + 1..s.pos].split(',').map(it.trim_space().trim_right('>').after(']'))
 							// if any typ is neither Type nor builtin, then the case is non-generic
-							for typ in typs {
-								if typ.len == 0
-									|| (!(typ[0].is_capital() && typ[1..].bytes().all(it.is_alnum()
-									|| it == `_`)) && typ !in ast.builtin_type_names) {
-									is_generic = false
-									break
-								}
-							}
+							typs.all(it.len > 0
+								&& ((it[0].is_capital() && it[1..].bytes().all(it.is_alnum()
+								|| it == `_`)) || it in ast.builtin_type_names))
 						} else {
-							is_generic = false
+							false
 						}
 						if is_generic {
 							return s.new_token(.gt, '', 1)

